_posts/2020-07-21-tips-and-tricks-for-running-strimzi-with-kubectl.md

```diff
@@ -0,0 +1,129 @@
+NAME                                DESIRED KAFKA REPLICAS   DESIRED ZK REPLICAS
+kafka.kafka.strimzi.io/my-cluster   3                        3
+
+NAME                                          PARTITIONS   REPLICATION FACTOR
+kafkatopic.kafka.strimzi.io/kafka-test-apps   3            3
+
+NAME                                 AUTHENTICATION   AUTHORIZATION
+kafkauser.kafka.strimzi.io/my-user   tls              simple
+```
+
+You can also combine this with other commands.
+For example while developing or testing new Strimzi features, I quite often need to delete all Strimzi resources.
+Doing it resource by resource would take a long time.
+But I can do it all in a single command:
+
+```
+$ kubectl delete $(kubectl get strimzi -o name)
+kafka.kafka.strimzi.io "my-cluster" deleted
+kafkatopic.kafka.strimzi.io "kafka-test-apps" deleted
+kafkauser.kafka.strimzi.io "my-user" deleted
+```
+
+The inner command - `kubectl get strimzi -o name` - will return the types and the resource names.
+This is enabled by the `-o name` option.
+
+```
+$ kubectl get strimzi -o name
+kafka.kafka.strimzi.io/my-cluster
+kafkatopic.kafka.strimzi.io/kafka-test-apps
+kafkauser.kafka.strimzi.io/my-user
+```
+
+And when this is passed into the `kubectl delete` command, it will delete all Strimzi resources.
+
+## Querying the status subresources
+
+We already saw how the `-o name` option can be used to get the output in the `type/name` format.
+There are also other values you can pass to the `-o` option.
+For example by using `-o yaml` you can get the output in YAML format.
+Or `-o json` will return it as JSON.
+You can see all the options in `kubectl get --help` or in the [`kubectl` docs](https://kubernetes.io/docs/reference/kubectl/overview/#formatting-output).
+
+One of the most useful options is `jsonpath`.
+It allows you to pass [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) expression which will be evaluated while querying the Kubernetes API.
+The JSONPath expression can be used to extract specific parts of the custom resource.
+For example, you can easily get the bootstrap address from the status of the Kafka custom resource and use it in your Kafka clients.
+You can do that in the following JSONPath expression - `{.status.listeners[?(@.type=="tls")].bootstrapServers}`:
+
+```
+$ kubectl get kafka my-cluster -o=jsonpath='{.status.listeners[?(@.type=="tls")].bootstrapServers}{"\n"}'
+my-cluster-kafka-bootstrap.myproject.svc:9093
+```
+
+The command above will find the `bootstrapServers` value of the `tls` listeners.
+By changing the type condition to `@.type=="external"` or `@.type=="plain"` you can also get the address of the other Kafka listeners.
+
+```
+$ kubectl get kafka my-cluster -o=jsonpath='{.status.listeners[?(@.type=="external")].bootstrapServers}{"\n"}'
+192.168.1.247:9094
+```
+
+You can similarly extract also all the other fields from any custom resource.
+
+## Conclusion
+
+`kubectl` seems to be on the first look a simple tool.
```

Comment:
```suggestion
`kubectl` on first appearance looks like a simple tool, but it has also some very powerful features, useful for everyday development.
I hope this blog post will help you to be more efficient when managing Strimzi with the `kubectl` tool.
```

Suggested simplification of the conclusion, I think mentioning custom resources again is redundant as you have explained that you can interact them much the same as other resources.